### PR TITLE
Support for CloudEvents format.

### PR DESCRIPTION
**Pull request by treeder**
Requires: https://github.com/fnproject/cli/pull/236

### CODE REVIEW

**Comment by crush-157** [comment]
The full example in README.md needs to be updated to look like the func.rb - i.e. test to see if `input.respond_to?(:fetch)`

Otherwise in the case of `--format default` when input is a string it's going to return `nil` unless that string contains `'name'`

**Comment by crush-157** [comment]
In context.rb, I'd suggest instead of...
```
    def protocol
      if cloud_event?
        return event['extensions']['protocol']
      end
      event['protocol']
    end
```
...use...
```
    def protocol
      cloud_event? ? event['extensions']['protocol'] : event['protocol']
    end
```
(and similarly for the other methods)

**Comment by crush-157** [comment]
And drop the `return` from `cloud_event?`

**Comment by crush-157** [comment]
In runner.rb you might want to look at the indentation of the cloudevent lambda.

File: lib/fdk/runner.rb [hunk 1]
```diff
@@ -8,19 +8,43 @@
...
+        # Skipping json parsing of body because it would already be a parsed map according to the format spec defined here: https://github.com/cloudevents/spec/blob/master/serialization.md#json
+        se = FDK.single_event(function: func, context: context, input: body)
+
+        # Respond with modified event
+        event['data'] = se
+        event['extensions']['protocol'] = {
+          headers: {
+              'Content-Type' => ['application/json']
+          },
+          'status_code' => 200,
```
Comment (rdallman) [inline lib/fdk/runner.rb]: need to find a way to make these more flexible in general going forward (i've some ideas), can discuss on 2.0 doc, this is OK for a time

**Pull request merged**